src/include/duckdb.h

```diff
@@ -2447,6 +2464,45 @@ Destroys the logical type and de-allocates all memory allocated for that type.
 */
 DUCKDB_API void duckdb_destroy_logical_type(duckdb_logical_type *type);
 
+/*!
+Creates a new custom type.
+
+* @return The custom type.
+*/
+DUCKDB_API duckdb_custom_type duckdb_create_custom_type();
```

Comment:
If we in the future want to support adding type modifiers that callback needs to be set somewhere outside of the logical type itself. We could of course add another function in that case: `duckdb_register_logical_type(duckdb_conneciton, duckdb_logical_type, duckdb_custom_type_callbacks)` but I thought the whole point of creating these struct types is so that we can add new settings later (without having to add a bunch of alternative top-level functions if that makes sense). In this case im good either way though.

Reply:
Yes, that's fair. Maybe we should then just remove `duckdb_custom_type_set_name` and have that be replaced by the LogicalType alias?